plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -768,6 +774,25 @@ private static SliceWriteFunction nvarcharWriteFunction()
                 .flatMap(dataCompression -> Enums.getIfPresent(DataCompression.class, dataCompression).toJavaUtil());
     }
 
+    private static Optional<DataCompression> getTableDataCompressionWithRetries(Handle handle, JdbcTableHandle table)
+    {
+        // DDL operations can take out locks against system tables causing the `getTableDataCompression` query to deadlock
+        // Retry if this deadlock occurs
+        RetryPolicy<Optional<DataCompression>> retryPolicy = new RetryPolicy<Optional<DataCompression>>()
+                .withMaxAttempts(3)
+                .handleIf(throwable ->
+                {
+                    Throwable rootCause = Throwables.getRootCause(throwable);
```

Comment:
Trying to test this was really challenging since I couldn't reproduce the test failure locally. Any tips would be appreciated.

Reply:
I think you caught an actual product issue since I believe end-users can run into this too.
We should fix the client instead of the test in this case I think.

Reply:
I agree - this is user-facing issue so retrying a test is not a solution.

Reply:
I was not able to reproduce the deadlocking of the system tables 🤷  

I *was* able to test this code correctly handles deadlocks by manually constructing a deadlocking situation with a sql script (similar to described here: https://stackoverflow.com/a/22825825)

And then modifying the `getTableDataCompression` query so that it gets deadlocked and killed.